src/main/java/bizbook/model/AddressBook.java

```diff
@@ -28,7 +30,9 @@ public class AddressBook implements ReadOnlyAddressBook {
         persons = new UniquePersonList();
     }
 
-    public AddressBook() {}
+    public AddressBook() {
+        pinnedPersons = new UniquePersonList();
+    }
```

Comment:
Would it work if you move the pinnedPersons to the non-static constructor above, that way, u do not need to setPinnedPerson(empty) in reset?